featurebyte/worker/task/batch_feature_create.py

```diff
@@ -0,0 +1,152 @@
+from featurebyte.service.feature import FeatureService
+from featurebyte.worker.task.base import BaseTask
+
+logger = get_logger(__name__)
+
+
+@contextmanager
+def set_environment_variable(variable: str, value: Any) -> Iterator[None]:
+    """
+    Set the environment variable within the context
+
+    Parameters
+    ----------
+    variable: str
+        The environment variable
+    value: Any
+        The value to set
+
+    Yields
+    ------
+    Iterator[None]
+        The context manager
+    """
+    previous_value = os.environ.get(variable)
+    os.environ[variable] = value
+
+    try:
+        yield
+    finally:
+        if previous_value is not None:
+            os.environ[variable] = previous_value
+        else:
+            del os.environ[variable]
+
+
+async def execute_sdk_code(catalog_id: ObjectId, code: str) -> None:
+    """
+    Activate the catalog and execute the code in server mode
+
+    Parameters
+    ----------
+    catalog_id: ObjectId
+        The catalog id
+    code: str
+        The SDK code to execute
+    """
+    # activate the correct catalog before executing the code
+    activate_catalog(catalog_id=catalog_id)
+
+    # execute the code
+    with set_environment_variable("SDK_EXECUTION_MODE", "SERVER"):
+        with concurrent.futures.ThreadPoolExecutor() as pool:
+            await asyncio.get_event_loop().run_in_executor(pool, exec, code)
+
+
+class BatchFeatureCreateTask(BaseTask):
+    """
+    Batch feature creation task
+    """
+
+    payload_class = BatchFeatureCreateTaskPayload
+
+    async def is_generated_feature_consistent(
+        self, document: FeatureModel, definition: str
+    ) -> bool:
+        """
+        Validate the generated feature against the expected feature
+
+        Parameters
+        ----------
+        document: FeatureModel
+            The feature document used to generate the feature definition
+        definition: str
+            Feature definition used at server side to generate the feature
+
+        Returns
+        -------
+        bool
+        """
+        # retrieve the saved feature & check if it is the same as the expected feature
+        feature_service: FeatureService = self.app_container.feature_service
+        feature = await feature_service.get_document(document_id=document.id)
+        generated_hash = feature.graph.node_name_to_ref[feature.node_name]
+        expected_hash = document.graph.node_name_to_ref[document.node_name]
+        is_consistent = definition == feature.definition and expected_hash == generated_hash
+        if not is_consistent:
+            # log the difference between the expected feature and the saved feature
+            logger.debug(
+                "Generated feature is not the same as the expected feature",
+                extra={
+                    "expected_hash": expected_hash,
+                    "generated_hash": generated_hash,
+                    "match_definition": definition == feature.definition,
```

Comment:
Would logging additional fields like feature name be useful for debugging?

Reply:
Will add feature name to the log info